src/simtools/ray_tracing/incident_angles.py

```diff
@@ -0,0 +1,592 @@
+        secondary_radius_m = [] if self.calculate_primary_secondary_angles else None
+        # Hit coordinates (meters), derived from cm columns
+        primary_hit_x_m = [] if self.calculate_primary_secondary_angles else None
+        primary_hit_y_m = [] if self.calculate_primary_secondary_angles else None
+        secondary_hit_x_m = [] if self.calculate_primary_secondary_angles else None
+        secondary_hit_y_m = [] if self.calculate_primary_secondary_angles else None
+
+        for parts in self._iter_data_rows(photons_file):
+            self._append_values(
+                parts,
+                col_idx,
+                focal,
+                primary,
+                secondary,
+                radius_m,
+                secondary_radius_m,
+                primary_hit_x_m,
+                primary_hit_y_m,
+                secondary_hit_x_m,
+                secondary_hit_y_m,
+            )
+
+        result = {"angle_incidence_focal_deg": focal}
+        if self.calculate_primary_secondary_angles:
+            result["angle_incidence_primary_deg"] = primary
+            result["angle_incidence_secondary_deg"] = secondary
+            result["primary_hit_radius_m"] = radius_m
+            result["secondary_hit_radius_m"] = secondary_radius_m
+            result["primary_hit_x_m"] = primary_hit_x_m
+            result["primary_hit_y_m"] = primary_hit_y_m
+            result["secondary_hit_x_m"] = secondary_hit_x_m
+            result["secondary_hit_y_m"] = secondary_hit_y_m
+        return result
+
+    def _discover_column_indices(self, photons_file):
+        """Return 0-based indices discovered from headers.
+
+        Returns a tuple: (focal_idx, primary_idx, secondary_idx,
+        primary_x_idx, primary_y_idx, secondary_x_idx, secondary_y_idx)
+
+        Defaults (1-based) are focal=26, primary=32, secondary=36,
+        primary X/Y = 29/30, secondary X/Y = 33/34.
+        """
+        indices = self._default_column_indices()
+
+        col_pat = re.compile(r"^\s*#\s*Column\s+(\d{1,4})\s*:(.*)$", re.IGNORECASE)
+        with photons_file.open("r", encoding="utf-8") as fh:
+            for raw in fh:
+                s = raw.strip()
+                if not s or not s.startswith("#"):
+                    continue
+                m = col_pat.match(s)
+                if not m:
+                    continue
+                num = int(m.group(1))
+                desc = m.group(2).strip().lower()
+                self._update_indices_from_header_desc(desc, num, indices)
+
+        return (
+            indices["focal"],
+            indices.get("primary"),
+            indices.get("secondary"),
+            indices.get("prim_x"),
+            indices.get("prim_y"),
+            indices.get("sec_x"),
+            indices.get("sec_y"),
+        )
+
+    def _default_column_indices(self):
+        """Return default 0-based indices matching legacy positions.
```

Comment:
What does 'legacy' mean in this context?

Reply:
These photon files are a bit messy as the column indices are changing between telescopes. This function here is to declare the initial values as is the case for the SST. Then we assign them new later if the column headers found are at a different index.